aiidalab/aiidalab
# Feature/aiidalab environment version

This changeset enables the specification of an app to environment compatibility map as part of the App metadata. The metadata is extended by the `aiidalab_environment_version` key which maps an app version to a compatible AiiDAlab environment version.

Here is a video demonstration as to how that looks in practice:
![ezgif-6-f5d069d1089f](https://user-images.githubusercontent.com/1441208/95607079-82576780-0a5b-11eb-8ed5-91518c33f920.gif)

To test this feature in your own instance:
1. Switch into AiiDAlab developer mode as described [here](https://github.com/aiidalab/aiidalab/wiki#development-flow-for-simultaneous-back-end-and-front-end-changes-aiidalab-and-home-app) or use the [aiidalab-dev](https://github.com/aiidalab/aiidalab-dev).

1. Switch the home app to this feature branch: https://github.com/csadorf/aiidalab-home/tree/feature/app-compatibility-with-environment
1. Download a local version of the [`apps_meta.json`](https://aiidalab.materialscloud.org/appsdata/apps_meta.json) file into your home directory and add the `aiidalab_environment_version` field.
1. Apply the following patch to aiidalab to ensure that the local registry is used:
```diff --git a/aiidalab/config.py b/aiidalab/config.py
index b33407b..4d60a5a 100644
--- a/aiidalab/config.py
+++ b/aiidalab/config.py
@@ -4,6 +4,6 @@ from os import getenv
 AIIDALAB_HOME = getenv('AIIDALAB_HOME', '/project')
 AIIDALAB_APPS = getenv('AIIDALAB_APPS', '/project/apps')
 AIIDALAB_SCRIPTS = getenv('AIIDALAB_SCRIPTS', '/opt')
-AIIDALAB_REGISTRY = getenv('AIIDALAB_REGISTRY', 'https://aiidalab.materialscloud.org/appsdata/apps_meta.json')
+AIIDALAB_REGISTRY = getenv('AIIDALAB_REGISTRY', 'file:///home/aiida/apps_meta.json')
 AIIDALAB_DEFAULT_GIT_BRANCH = getenv('AIIDALAB_DEFAULT_GIT_BRANCH', 'master')
 AIIDALAB_ENVIRONMENT_VERSION = getenv('AIIDALAB_ENVIRONMENT_VERSION', '1.0.0')
```
